src/js/tree_row_header.js

```diff
@@ -38,6 +38,11 @@ function _tree_header_levels(path, is_open, is_leaf) {
 export function tree_header(td, path, types, is_leaf, is_open) {
     const type = types[path.length - 1];
     const name = path.length === 0 ? "TOTAL" : path[path.length - 1];
+
+    if (path.length && path[path.length - 1].endsWith("/")) {
```

Comment:
This doesn't belong here.  I'd like to hear a proposal on how to handle tree rendering as part of the data model API, rather than bash this implementation into submission, especially as there are tree forms the current renderer does not support.

Reply:
I agree, although this was nicely quick and easy. I'll think about this. A couple of idea off the top of my head:

1. row, col, cell, etc metadata that can be explicitly provided by the data model
2. make the data model itself hierarchical

I played around with 2. a bit while I was originally implementing the tree for file_browser.html. It's probably not a great idea, since it completely screws up the concept/utility of row index, but maybe there's a way to do an appropriate nested data structure here?